src/commands/to.rs

```diff
@@ -0,0 +1,181 @@
+use crate::MsgPackPlugin;
+use nu_plugin::{EngineInterface, EvaluatedCall, SimplePluginCommand};
+use nu_protocol::{
+    record, Category, Example, LabeledError, Signature, Span, SyntaxShape, Type, Value,
+};
+
+pub struct ToMsgpack;
+
+impl SimplePluginCommand for ToMsgpack {
+    type Plugin = MsgPackPlugin;
+
+    fn name(&self) -> &str {
+        "to msgpack"
+    }
+
+    fn usage(&self) -> &str {
+        "Converts data to msgpack."
+    }
+    fn signature(&self) -> Signature {
+        Signature::build(self.name())
+            .named(
+                "brotli",
+                SyntaxShape::Int,
+                "Brotli Encoder Mode (0 - 11)",
+                Some('b'),
+            )
+            .category(Category::Formats)
+            .input_output_type(Type::Any, Type::Binary)
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["msgpack", "plugins"]
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                example: "{ greeting: 'hello world' } | to msgpack",
+                description: "Encode msgpack",
+                result: Some(Value::test_binary(b"\x81\xA8\x67\x72\x65\x65\x74\x69\x6E\x67\xAB\x68\x65\x6C\x6C\x6F\x20\x77\x6F\x72\x6C\x64")),
+            },
+            Example {
+                example: "{ greeting: 'hello world this is a long string' } | to msgpack --brotli 9",
+                description: "Encode msgpack and compress strings with brotli (level 9)",
+                result: None,
+            },
+            Example {
+                example: "{ greeting: 'hello world this is a long string' } | to msgpack --brotli 9 | from msgpack --brotli",
+                description: "Encode and decode msgpack with brotli compressed strings",
+                result: Some(Value::test_record(record! {
+                    "greeting" => Value::test_string("hello world this is a long string")
+                })),
+            },
+            Example {
+                example: "{ hello: world } | save helloworld.msgpack",
+                description: "Save msgpack to a file",
+                result: None,
+            }
+        ]
+    }
+
+    fn run(
+        &self,
+        _plugin: &MsgPackPlugin,
+        _engine: &EngineInterface,
+        call: &EvaluatedCall,
+        input: &Value,
+    ) -> Result<Value, LabeledError> {
+        let compression: Option<i32> = call.get_flag::<i64>("compression")?.map(|c| c as i32);
```

Comment:
Seems like this should be the parameter name since you changed it from compression, right?
```suggestion
        let compression: Option<i32> = call.get_flag::<i64>("brotli")?.map(|c| c as i32);
```

Reply:
oops, yep, good catch